redbot/cogs/trivia/session.py

```diff
@@ -98,8 +98,24 @@ def start(cls, ctx, question_list, settings):
         session = cls(ctx, question_list, settings)
         loop = ctx.bot.loop
         session._task = loop.create_task(session.run())
+        session._task.add_done_callback(session._error_handler)
         return session
 
+    def _error_handler(self, fut):
+        """Catches errors in the session task."""
+        try:
+            fut.result()
+        except asyncio.CancelledError:
+            pass
+        except Exception as exc:
+            LOG.error("A trivia session has encountered an error.\n", exc_info=exc)
+            asyncio.create_task(
+                self.ctx.send(
+                    "An unexpected error occurred in the trivia session.\nCheck your console or logs for details."
+                )
+            )
+            asyncio.create_task(self.end_game())
```

Comment:
I don't think you should call end_game as it will pay out the winner. We don't pay out the winner on a timeout, we probably shouldn't pay them out on an error. Best just call `self.stop()`, which has the bonus of being a synchronous function.

Moreover, I think the current handler leaves the trivia session object in the trivia cog's cache. `TriviaSession.stop()` takes care of that.

Reply:
Although `end_game` does call `stop` at the end, I agree it shouldn't give points regardless.